File: demographics/src/main/java/com/medilabo/demographics/model/Patient.java
package com.medilabo.demographics.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.OffsetDateTime;

@Entity
@Table(name = "patients")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "date_of_birth", nullable = false)
    private LocalDate dateOfBirth;

    @Column(name = "gender", nullable = false, length = 1)
    private String gender;

    @Column(name = "address")
    private String address;

    @Column(name = "phone")
    private String phone;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private OffsetDateTime createdAt;
}

File: frontend/src/types/patient.ts
export interface Patient {
  id: number
  lastName: string
  firstName: string
  dateOfBirth: string // ISO date string: 'YYYY-MM-DD'
  gender: 'M' | 'F'
  address?: string
  phone?: string
  createdAt?: string
}

export interface PatientFormValues {
  lastName: string
  firstName: string
  dateOfBirth: string
  gender: 'M' | 'F' | ''
  address?: string
  phone?: string
}
